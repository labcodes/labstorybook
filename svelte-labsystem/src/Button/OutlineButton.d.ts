import { IconTypes, LocalSvelteComponent } from "../shared";

export interface IOutlineButtonProps {
  text: string,
  skin?: "light"
  | "dark",
  icon?: IconTypes,
  size?: "normal" | "small" | "large",
  disabled?: boolean,
  fullWidth?: boolean,
}

declare class OutlineButton extends LocalSvelteComponent<IOutlineButtonProps> { }
export default OutlineButton;
