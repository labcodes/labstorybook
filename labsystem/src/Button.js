import React from "react";
import PropTypes from "prop-types";
// import { isUndefined } from "loadash";

export default class Button extends React.Component {
  static propTypes = {
    disabled: PropTypes.bool,
    type: PropTypes.string,
    size: PropTypes.string,
    
  };

  static defaultProps = {
    disabled: false,
    type: undefined,
    size: undefined,

  };

  render() {
    const { disabled, type, size } = this.props;
    return (
      <button className={`btn btn--${type} btn--${size} `} disabled={disabled}>{this.props.children}</button> 
    );
  }
}
