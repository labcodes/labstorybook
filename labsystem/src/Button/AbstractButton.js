import React from "react";
import PropTypes from "prop-types";
import { isUndefined } from "lodash";
import Icon from "../Icon";

export default class AbstractButton extends React.Component {
  static propTypes = {
    type: PropTypes.oneOf(["button", "submit", "reset"]),
    text: PropTypes.string.isRequired,
    variant: PropTypes.oneOf(["default", "outline", "text"]),
    skin: PropTypes.oneOf([
      "light",
      "dark",
      "warning",
      "destructive",
      "warning-invert",
      "destructive-invert",
      "confirmation-invert",
    ]),
    icon: PropTypes.string,
    size: PropTypes.oneOf(["normal", "small", "large"]),
    disabled: PropTypes.bool,
    handleClick: PropTypes.func,
  };

  static defaultProps = {
    type: "button",
    variant: "default",
    skin: undefined,
    icon: undefined,
    size: "normal",
    disabled: false,
    handleClick: undefined,
  };

  icon = () => {
    const { icon } = this.props;

    /*
    Set icon color to be the icon color of default button.
    Icon color changes according to button type and exception on _buttons.scss
    */
    return icon ? (
      <Icon type={icon} color="white" size="petit" className="btn__icon" />
    ) : (
      ""
    );
  };

  handleOnClick = (e) => {
    const { handleClick } = this.props;
    if (!isUndefined(handleClick)) {
      handleClick(e);
    }
  };

  render() {
    const { type, text, variant, skin, size, disabled } = this.props;
    return (
      <button
        // eslint-disable-next-line react/button-has-type
        type={type}
        className={`btn btn--${variant} btn--${skin} btn--${size}`}
        disabled={disabled}
        onClick={this.handleOnClick}
      >
        {this.icon()}
        {text}
      </button>
    );
  }
}
