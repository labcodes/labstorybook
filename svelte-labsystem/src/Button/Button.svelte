<script>
  import Icon from '../Icon.svelte';

  export let text;
  export let skin = undefined;
  export let type = undefined;
  export let icon = undefined;
  export let size = "normal";
  export let disabled = false;
  export let fullWidth = false;
</script>


<button
  {type}
  class={`lab-btn` + ` lab-btn--default lab-btn--${size}` + `${skin ? ` lab-btn--${skin}` : ''}` + `${fullWidth ? ` lab-btn--block` : ''}`}
  on:click
  {disabled}>
  {#if icon}
    <Icon type={icon} color="white" size="petit" class="lab-btn__icon" />
  {/if}
  {text}
</button>
