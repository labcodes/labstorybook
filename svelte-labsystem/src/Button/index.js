import Button from "./Button.svelte";

export { default as Button } from "./Button.svelte";
export { default as OutlineButton } from "./OutlineButton.svelte";
export { default as TextButton } from "./TextButton.svelte";
export default Button;
