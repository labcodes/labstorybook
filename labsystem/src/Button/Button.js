import React from "react";
import PropTypes from "prop-types";
import AbstractButton from "./AbstractButton";

export default class Button extends React.Component {
  static propTypes = {
    /** This is the button label. */
    text: PropTypes.string.isRequired,
    /** Sets a special color skin to the button. */
    skin: PropTypes.oneOf([
      "",
      "light",
      "dark",
      "warning",
      "destructive",
      "warning-invert",
      "destructive-invert",
      "confirmation-invert",
    ]),
    /** Sets the icon related to the button label. Default state: no icon. */
    icon: PropTypes.string,
    /** Sets the button's height. Small = 32px, Normal = 40px, Large = 48px. */
    size: PropTypes.oneOf(["normal", "small", "large"]),
    /** Makes the button disabled, cancelling the onClick handler. */
    disabled: PropTypes.bool,
    /** Action to be executed when the button is clicked. */
    onClick: PropTypes.func,
    /** Makes the button expand to its container's full width. */
    fullWidth: PropTypes.bool,
  };

  static defaultProps = {
    skin: undefined,
    icon: undefined,
    size: "normal",
    disabled: false,
    onClick: undefined,
    fullWidth: false,
  };

  render() {
    const { text, skin, icon, size, disabled, onClick, fullWidth } = this.props;
    return (
      <AbstractButton
        variant="default"
        text={text}
        skin={skin}
        icon={icon}
        size={size}
        disabled={disabled}
        onClick={onClick}
        fullWidth={fullWidth}
      />
    );
  }
}
