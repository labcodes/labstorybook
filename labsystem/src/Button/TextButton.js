import React from "react";
import PropTypes from "prop-types";
import AbstractButton from "./AbstractButton";

export default class TextButton extends React.Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
    skin: PropTypes.oneOf(["", "light", "dark"]),
    icon: PropTypes.string,
    size: PropTypes.oneOf(["normal", "small", "large"]),
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
    fullWidth: PropTypes.bool,
  };

  static defaultProps = {
    skin: undefined,
    icon: undefined,
    size: "normal",
    disabled: false,
    onClick: undefined,
    fullWidth: false,
  };

  render() {
    const { text, skin, icon, size, disabled, onClick, fullWidth } = this.props;
    return (
      <AbstractButton
        variant="text"
        text={text}
        skin={skin}
        icon={icon}
        size={size}
        disabled={disabled}
        onClick={onClick}
        fullWidth={fullWidth}
      />
    );
  }
}
