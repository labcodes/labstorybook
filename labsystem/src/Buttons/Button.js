import React from "react";
import PropTypes from "prop-types";
import AbstractButton from "./AbstractButton";

export default class Button extends React.Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
    skin: PropTypes.oneOf([
      "light",
      "dark",
      "warning",
      "destructive",
      "warning-invert",
      "destructive-invert",
      "confirmation-invert",
    ]),
    icon: PropTypes.string,
    size: PropTypes.oneOf(["normal", "small", "large"]),
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    onKeyUp: PropTypes.func,
    fullWidth: PropTypes.bool,
  };

  static defaultProps = {
    skin: undefined,
    icon: undefined,
    size: "normal",
    disabled: false,
    onClick: undefined,
    onFocus: undefined,
    onBlur: undefined,
    onKeyUp: undefined,
    fullWidth: false,
  };

  render() {
    const {
      text,
      skin,
      icon,
      size,
      disabled,
      onClick,
      fullWidth,
      onFocus,
      onBlur,
      onKeyUp,
    } = this.props;
    return (
      <AbstractButton
        variant="default"
        text={text}
        skin={skin}
        icon={icon}
        size={size}
        disabled={disabled}
        onClick={onClick}
        onFocus={onFocus}
        onBlur={onBlur}
        onKeyUp={onKeyUp}
        fullWidth={fullWidth}
      />
    );
  }
}
