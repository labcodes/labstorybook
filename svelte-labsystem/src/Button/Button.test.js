import { render } from "@testing-library/svelte";
import Button from "./Button.svelte";

describe("Button", () => {
  it("renders with base props", async () => {
    const { container } = render(Button, { text: "test" });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing disabled as true", async () => {
    const { container, getByText } = render(Button, {
      text: "test",
      disabled: true,
    });

    expect(container).toMatchSnapshot();
    expect(getByText("test").disabled).toBeTruthy();
  });

  it("renders as expected when passing a warning skin", async () => {
    const { container } = render(Button, { text: "test", skin: "warning" });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing a small size", async () => {
    const { container } = render(Button, { text: "test", size: "small" });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing a wallet icon", async () => {
    const { container } = render(Button, { text: "test", icon: "wallet" });
    expect(container).toMatchSnapshot();
  });
});
