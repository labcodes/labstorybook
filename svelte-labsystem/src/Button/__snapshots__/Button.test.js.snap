// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders as expected when passing a small size 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--default lab-btn--small"
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`Button renders as expected when passing a wallet icon 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--default lab-btn--normal"
    >
      <span
        class="lab-icon lab-icon--wallet lab-icon--white lab-icon--petit lab-btn__icon"
      />
       
      test
    </button>
  </div>
</body>
`;

exports[`Button renders as expected when passing a warning skin 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--default lab-btn--normal lab-btn--warning"
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`Button renders as expected when passing disabled as true 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--default lab-btn--normal"
      disabled=""
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`Button renders with base props 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--default lab-btn--normal"
    >
       
      test
    </button>
  </div>
</body>
`;
